sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/configuration/AADOAuth2ClientConfiguration.java

```diff
@@ -28,60 +31,146 @@
  * </p>
  */
 @Configuration(proxyBeanMethods = false)
-@Conditional(ClientRegistrationCondition.class)
 public class AADOAuth2ClientConfiguration {
 
     /**
-     * Declare ClientRegistrationRepository bean.
-     *
-     * @param properties the AAD authentication properties
-     * @return ClientRegistrationRepository bean
+     * OAuth2 client configuration for AAD.
      */
-    @Bean
-    @ConditionalOnMissingBean
-    public ClientRegistrationRepository clientRegistrationRepository(AADAuthenticationProperties properties) {
-        return new AADClientRegistrationRepository(properties);
+    @Configuration(proxyBeanMethods = false)
+    @Conditional(ClientRegistrationCondition.class)
+    public static class OAuth2ClientRepositoryConfiguration {
+
+        /**
+         * Declare ClientRegistrationRepository bean.
+         *
+         * @param properties the AAD authentication properties
+         * @return ClientRegistrationRepository bean
+         */
+        @Bean
+        @ConditionalOnMissingBean
+        public ClientRegistrationRepository clientRegistrationRepository(AADAuthenticationProperties properties) {
+            return new AADClientRegistrationRepository(properties);
+        }
+
+        /**
+         * Declare OAuth2AuthorizedClientRepository bean.
+         *
+         * @return OAuth2AuthorizedClientRepository bean
+         */
+        @Bean
+        @ConditionalOnMissingBean
+        public OAuth2AuthorizedClientRepository oAuth2AuthorizedClientRepository() {
+            return new JacksonHttpSessionOAuth2AuthorizedClientRepository();
+        }
     }
 
     /**
-     * Declare OAuth2AuthorizedClientRepository bean.
-     *
-     * @return OAuth2AuthorizedClientRepository bean
+     * Web application scenario, OAuth2AuthorizedClientManager configuration for AAD.
      */
-    @Bean
-    @ConditionalOnMissingBean
-    public OAuth2AuthorizedClientRepository oAuth2AuthorizedClientRepository() {
-        return new JacksonHttpSessionOAuth2AuthorizedClientRepository();
+    @Configuration(proxyBeanMethods = false)
+    @Conditional(WebApplicationWithoutResourceServerCondition.class)
+    public static class WebApplicationOAuth2AuthorizedClientManagerConfiguration {
```

Comment:
nit: Rename to `WebApplicationWithoutResourceServerOAuth2AuthorizedClientManagerConfiguration`